### DISCUSSION TITLE
How to use custom activity attribute  Liquid expressions

### DISCUSSION THREAD

**Discussion by Gavin0912** [elsa-workflows/elsa-core#1220] [Q&A]
![image](https://user-images.githubusercontent.com/20514838/124534875-b7a4b080-de47-11eb-9621-550c29e4bdcf.png)

**Comment by sfmskywalker** [comment]
When using the designer, you will see that you can configure the **UserName** field to be written in liquid syntax.

Example:

![](https://elsa-workflows.github.io/elsa-core/docs/assets/guides/guides-recurring-tasks-animation-1.gif)

**Reply by Gavin0912** [reply]
Is the WorkflowExpression method rewritten in the current version?

**Reply by sfmskywalker** [reply]
I'm not sure what you mean, but the liquid handling is largely the same as in Elsa 1.

**Comment by Gavin0912** [comment]
I may not express it clearly. What I mean is that when I monitor the context after using the liquid expression on the designer, the expression does not translate the normal value. Do I need to configure other things?
![image](https://user-images.githubusercontent.com/20514838/124710869-f2821380-df2f-11eb-80e3-cae502d46706.png)
![image](https://user-images.githubusercontent.com/20514838/124711005-21988500-df30-11eb-97a4-14cb226d6915.png)
I'm not very clear why the value is not taken in this place. I see that the previous version has an expression to notify the handler. Do I need to add this place in the new version?

**Reply by sfmskywalker** [reply]
When using Liquid together with custom/complex types such as `User`, make sure you allow-list these types with the Liquid engine as described here: https://elsa-workflows.github.io/elsa-core/docs/next/expressions/expressions-liquid#allow-listing-a-specific-type

**Reply by Gavin0912** [reply]
I am a beginner, and I did it in accordance with the document method, but I still cannot get the value of the expression in my custom activity node. I don’t know where the problem lies.
![image](https://user-images.githubusercontent.com/20514838/124871050-24f64400-dff6-11eb-8674-13b0cb24d8b7.png)
![image](https://user-images.githubusercontent.com/20514838/124871105-38091400-dff6-11eb-8ab1-ddc3e152405c.png)
![image](https://user-images.githubusercontent.com/20514838/124871181-4fe09800-dff6-11eb-9670-0cd8f77cc7fe.png)
![image](https://user-images.githubusercontent.com/20514838/124871346-85858100-dff6-11eb-8755-1a5097bb47ca.png)

**Reply by sfmskywalker** [reply]
Your Liquid expression is `{{ Input.RegistrationModel.Name }}`. This works if you are sending some object containing a `RegistrationModel` property as input.

Can you show us where you are sending this input?

**Reply by Gavin0912** [reply]
![image](https://user-images.githubusercontent.com/20514838/124873282-d8f8ce80-dff8-11eb-8087-5b4a1c930bab.png)
Yes, I have changed the name of RegistrationModel to UserModelDto. I also changed the name to this name on the flowchart. I started the process by sending a signal.
![image](https://user-images.githubusercontent.com/20514838/124873931-ae5b4580-dff9-11eb-922d-3cefbf642d3b.png)

**Reply by sfmskywalker** [reply]
Ok good. Then try these expressions instead:

`{{ Input.Name }}`
`{{ Input.Email }}`

etc.

You are passing in `model` (which is an instance of `UserModelDto`) into the workflow.
This means that in liquid, `Input` refers to exactly that object.

**Reply by Gavin0912** [reply]
Yes, I can successfully get the value using {{ Input.Name }} this expression, thank you!
